### PR TITLE
Added basic get/post for feed and items

### PR DESCRIPTION
**Pull request by mfasquel**

### CODE REVIEW

File: src/main/kotlin/fsrss/FeedController.kt [hunk 1]
```diff
@@ -1,17 +1,63 @@
...
+    fun readFeed(@PathVariable feed: String) : String {
+        val path = Path.of(path, feed)
+        if (Files.exists(path)) {
+            return FeedHandler().createRssFeed(path.toFile())
+        } else {
+            throw HttpClientErrorException(HttpStatus.BAD_REQUEST, "Feed does not exist")
+        }
+    }
+
+    @RequestMapping("/feeds/{feed}")
```
Comment (wimgoeman) [inline src/main/kotlin/fsrss/FeedController.kt]:
```suggestion
    @PostMapping("/feeds/{feed}")
```

File: src/main/kotlin/fsrss/FeedController.kt [hunk 2]
```diff
@@ -1,17 +1,63 @@
...
+            Files.createDirectory(path)
+            val metaFile = File("$path/meta.json")
+            metaFile.createNewFile()
+            metaFile.writeText(jsonMapper.writeValueAsString(meta))
+        } else {
+            throw HttpClientErrorException(HttpStatus.BAD_REQUEST, "Feed already exist")
+        }
+    }
+
+    @RequestMapping("/feeds/{feed}/item")
```
Comment (wimgoeman) [inline src/main/kotlin/fsrss/FeedController.kt]: We could probably use the "channel" level as well, so /feeds/{feed}/channels/{channel}/items/{item}

File: src/main/kotlin/fsrss/FeedController.kt [hunk 3]
```diff
@@ -1,17 +1,63 @@
...
+            throw HttpClientErrorException(HttpStatus.BAD_REQUEST, "Feed does not exist")
+        }
+    }
+
+    @RequestMapping("/feeds/{feed}")
+    @ResponseStatus(HttpStatus.OK)
+    fun createFeed(@PathVariable feed: String, @RequestBody body: String) {
+        val path = Path.of(path + feed)
+        if (!Files.exists(path)) {
+            val meta: FeedMetaData = jsonMapper.readValue(body)
```
Comment (wimgoeman) [inline src/main/kotlin/fsrss/FeedController.kt]:
```suggestion
            val meta = jsonMapper.readValue(body)
```

File: src/main/kotlin/fsrss/FeedController.kt [hunk 4]
```diff
@@ -1,17 +1,63 @@
...
+import org.springframework.web.bind.annotation.*
+import org.springframework.web.client.HttpClientErrorException
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
 
 
 @Controller
 class FeedController {
+    val path: String = System.getenv("RSS_FEEDS_PATH") ?: "/"
```
Comment (wimgoeman) [inline src/main/kotlin/fsrss/FeedController.kt]:
```suggestion
    val path: String = System.getenv("RSS_FEEDS_PATH") ?: "/data"
```
To match my just written readme. Also easier for much needed security at some point I guess.

File: src/main/kotlin/fsrss/FeedController.kt [hunk 5]
```diff
@@ -1,17 +1,63 @@
...
+    }
+
+    @RequestMapping("/feeds/{feed}")
+    @ResponseStatus(HttpStatus.OK)
+    fun createFeed(@PathVariable feed: String, @RequestBody body: String) {
+        val path = Path.of(path + feed)
+        if (!Files.exists(path)) {
+            val meta: FeedMetaData = jsonMapper.readValue(body)
+            Files.createDirectory(path)
+            val metaFile = File("$path/meta.json")
```
Comment (wimgoeman) [inline src/main/kotlin/fsrss/FeedController.kt]: Use nio instead?

File: src/main/kotlin/fsrss/FeedController.kt [hunk 6]
```diff
@@ -1,17 +1,63 @@
...
+    }
+
+    @RequestMapping("/feeds/{feed}/item")
+    @ResponseStatus(HttpStatus.OK)
+    fun addItem(@PathVariable feed:String, @RequestBody body: String) {
+        val path = Path.of(path, feed)
+        if (Files.exists(path)) {
+            val item: Item = jsonMapper.readValue(body)
+            val sanitizedTitle: String = item.title.filter{it.isLetterOrDigit()}.toLowerCase()
+            val itemFile = File(path.toFile(), sanitizedTitle)
```
Comment (wimgoeman) [inline src/main/kotlin/fsrss/FeedController.kt]: nio here as well?

File: src/main/kotlin/fsrss/FeedController.kt [hunk 7]
```diff
@@ -1,17 +1,63 @@
...
+            if (!itemFile.exists())  {
+                itemFile.createNewFile()
+                itemFile.writeText(jsonMapper.writeValueAsString(item))
+            } else {
+                throw HttpClientErrorException(HttpStatus.BAD_REQUEST, "Item already exist")
```
Comment (wimgoeman) [inline src/main/kotlin/fsrss/FeedController.kt]: I'm not at all sure about this, but I think you can set the return type to Response, and return Response.ko("error goes here")

File: src/main/kotlin/fsrss/FeedController.kt (hunk 2 above)
Comment (mfasquel) [reply src/main/kotlin/fsrss/FeedController.kt]: There can only by one channel inside a RSS feed as far as I know. There is nothing specific to them

File: src/main/kotlin/fsrss/FeedController.kt (hunk 3 above)
Comment (mfasquel) [reply src/main/kotlin/fsrss/FeedController.kt]: the jsonMapper need info on the type it targets. Either for the val declaration or in the function call

**Review by wimgoeman — APPROVED** [review]

**Pull request merged**